### PR TITLE
Include timings in trace messages when `profiler` is true

### PR DESCRIPTION
**Pull request by Kha**
Rudimentary implementation. Would be interesting to have an "expand hot path" context menu item.

image

### CODE REVIEW

**Comment by Kha** [comment]
!bench

File: src/Lean/Util/Trace.lean [hunk 1]
```diff
@@ -150,18 +150,27 @@ private def addTraceNode (oldTraces : PersistentArray TraceElem)
...
+def withTraceNode [MonadExcept ε m] [MonadLiftT BaseIO m] (cls : Name) (msg : Except ε α → m MessageData) (k : m α)
     (collapsed := true) : m α := do
   if !(← isTracingEnabledFor cls) then
     k
   else
     let ref ← getRef
     let oldTraces ← getResetTraces
-    let res ← observing k
-    addTraceNode oldTraces cls ref (← msg res) collapsed
+    let (res, secs) ← withSeconds <| observing k
```
Comment (leodemoura) [inline src/Lean/Util/Trace.lean:166]: `withSeconds` is being executed even when `profiler` is set to false. Have you checked the impact on performance?

Comment (leodemoura) [reply src/Lean/Util/Trace.lean:166]: I see you tried `!bench!` last week, but it seems it did not work.

**Comment by Kha** [comment]
!bench

**Comment by Kha** [comment]
!bench

**Comment by Kha** [comment]
!bench

**Pull request merged**